agenta-web/src/lib/helpers/openapi_parser.ts

```diff
@@ -46,6 +37,10 @@ export const openAISchemaToParameters = (schema: GenericObject): Parameter[] =>
                 maximum: param["maximum"] ? param.maximum : 1,
                 required: !!schema.components.schemas[bodySchemaName]?.required?.includes(name),
             }
+            // above should be refactored to include only appropriate fields per x-parameter type
+            if (parameter.type === "grouped_choice") {
+                parameter.choices = param["choices"]
```

Comment:
why not add it to the above?

```
choice: param["choices"] ? param.choices : null
```

Reply:
because we will end up with useless keys not needed for some parameters.
Example we don't need maximum and minimum for grouped_choice.
we don't need enum for file_url

Reply:
Who cares if we have useless empty keys for some parameters? I find the code much cleaner the other way.
In any case, it does not make sense to tackle the same problem (not all parameters are relevant) in two different ways (once by setting default in the lines above, and once with an if).

Please either refactor it to have it your way, or use the same standard for the rest. But using your standard for your part, and adding a comment that above should be refactored is not good in my opinion. I don't think we should add comments to refactor 5 lines of code if it can be done in the same PR.